### PR TITLE
chore(mobile): Expo 51 + deps update

### LINKED ISSUE #12600: Update expo and related dependencies

### PR DESCRIPTION
**Pull request by Nodonisko**
## Related Issue

Resolve [#12600](https://github.com/trezor/trezor-suite/issues/12600)

### CODE REVIEW

File: suite-native/app/package.json [hunk 1]
```diff
@@ -73,44 +73,43 @@
...
-        "expo-image": "1.10.6",
-        "expo-linear-gradient": "12.7.1",
-        "expo-localization": "14.8.3",
-        "expo-navigation-bar": "2.8.1",
-        "expo-secure-store": "12.8.1",
-        "expo-splash-screen": "0.26.4",
-        "expo-status-bar": "~1.11.1",
-        "expo-system-ui": "2.9.3",
-        "lottie-react-native": "6.6.0",
+        "expo": "^51.0.9",
```
Comment (matejkriz) [inline suite-native/app/package.json]: @Nodonisko Why did you switch from the exact version to the carret range everywhere?

Comment (Nodonisko) [reply suite-native/app/package.json]: default behavior of yarn up command, don't want to fight it

Comment (Nodonisko) [reply suite-native/app/package.json]: It could be configured so I will update it to exact versions again

Comment (matejkriz) [reply suite-native/app/package.json]: I belive there is no real difference for us, I was just curious...

**Comment by Nodonisko** [comment]
@SocketSecurity ignore-all

**Review by matejkriz — APPROVED** [review]
🎉 

image

**Pull request merged**